File: PersonnelDepartment/Domain/Posts/Post.cs
namespace PersonnelDepartment.Domain.Posts;

public class Post
{
    public Guid Id { get; }
    public Guid DepartmentId { get; }
    public String Name { get; }
    public Int32 Salary { get; }

    public Post(Guid id, Guid departmentId, String name, Int32 salary)
    {
        Id = id;
        DepartmentId = departmentId;
        Name = name;
        Salary = salary;
    }
}

File: PersonnelDepartment/ClientApp/src/domain/posts/post.ts
export class Post {
    constructor(
        public id: string,
        public departmentId: string,
        public name: string,
        public salary: number
    ) { }
}

export function mapToPost(data: any): Post {
    return new Post(data.id, data.departmentId, data.name, data.salary);
}

export function mapToPosts(data: any[]): Post[] {
    return data.map(mapToPost);
}